### ISSUE TITLE
Assembled reactor breaks after re-login if interacted via block reader from Integrated Dynamics

### ISSUE THREAD

**Issue by Patryk122PL** [BiggerSeries/BiggerReactors#21]
Hello,

I found out while tinkering with reactor that if you interact with reactor controller via block reader from mod Integrated Dynamics (e.g read NBT data) the reactor multiblock structure will not form after re-login.

At first I assumed it was an issue with ID mod (maybe it's both?), however further testing seems to indicate that this is a problem with multiblock check logic, because even if I removed block reader the issue persisted like somehow NBT structure was broken inside controller's tile entity.

Also multiple re-login attempts seem to cause reactor multiblock structure to sometimes assemble and other times not to.

Removing ID helps (that's why I thought it was ID problem), but if I keep Integrated Dynamics and won't interact with reactor controller then everything works fine. So it only breaks after this two mods interact.

PS. I attached some screenshots and marked with red frame the block that reactor "thinks" it's air. Also if it helps the screen from ID shows fuel level read from reactor's NBT data.

![Zrzut ekranu 2021-07-23 o 08 14 11](https://user-images.githubusercontent.com/9610133/126747229-0281e067-b251-46a8-811d-54300d1d5db1.png)
![Zrzut ekranu 2021-07-23 o 08 17 58](https://user-images.githubusercontent.com/9610133/126747250-47b52106-ec03-4259-bfb5-343cba6b87f7.png)

**Comment by RogueLogix** [comment]
What version? Also get the Phosphophyllite version.

**Comment by Patryk122PL** [comment]
Oh sorry.. I should've include this in OP..

MC 1.16.5, forge 1.16.5-36.1.65

biggerreactors-1.16.5-0.5.0-beta.3
phosphophyllite-1.16.5-0.5.0

IntegratedDynamics-1.16.5-1.9.0
CommonCapabilities-1.16.5-2.7.0
CyclopsCore-1.16.5-1.11.7
integratednbt-1.16.4-1.4.2
IntegratedTunnels-1.16.5-1.8.4

**Comment by RogueLogix** [comment]
i doubt it will change anything, but try the latest versions

**Comment by Patryk122PL** [comment]
Somethings got changed that's for sure. Simulation data doesn't exist in NBT now, so I extracted Z pos value and it still breaks, but like I said in OP.. Reactor seems to work, after re-login - breaks, then after another re-login works, and so on and so forth. 

Like so:
![Zrzut ekranu 2021-07-23 o 09 32 57](https://user-images.githubusercontent.com/9610133/126751333-a8ac6670-64f0-486a-bca3-63f555b829c7.png)

**Comment by RogueLogix** [comment]
One change in 0.5.1 was only a single block saves the simulation data, instead of all of them. Currently working on getting a 1.17 version,  once I get something basic, ill pull up my 1.16 env and look into this.

**Comment by Patryk122PL** [comment]
Thanks, I appreciate that. 

Btw. as a side note - you should update your mod's screenshots on curseforge. I checked this mod earlier this year and didn't like the textures and lack of connected textures support so I stayed with extreme reactors which I've known back from the 1.12.2 days..

Lately I decided to give this mod another shot (despite pictures and because I'm looking for an alternative to ER) and was pleasantly surprised with not only amazing new textures but also connected textures support.

What I'm trying to say is this old pictures really disservice this otherwise great mod.

**Comment by RogueLogix** [comment]
updating those is on the todo list

**Comment by RogueLogix** [comment]
unable to reproduce in dev

**Comment by Patryk122PL** [comment]
I recreated the same reactor setup in new instance with necessary mods only and for me this issue still persists. I'm attaching my test instance below.

The reactor is at X,Y pos: 0.

[BiggerReactorTest.zip](https://github.com/BiggerSeries/BiggerReactors/files/6911237/BiggerReactorTest.zip)

**Comment by Patryk122PL** [comment]
I've tested numerous of blocks from Integrated Dynamics and it seems that most of "working blocks" (tile entities) break the reactor, except squeezer and drying basin afaik.. Even single pice of logic cable next to any reactor's block is enough to cause this bug.

I thought maybe any TE that touch the reactor will break it, but none of vanilla stuff did.

Update: I've placed single pice of logic cable 15 blocks from reactor's center and it breaks, but if I do it 16 blocks from center then it works normally (perhaps it breaks within a chunk?).

Also after repeated quits and re-login I've noticed that sometimes (like every 10-15 attempt) the reactor not only breaks visually but also it's NBT data gets completely wipeout.. everything - produced FE, fuel inside fuel cells and reactor state switches from active to offline.

This looks like it's more of Integrated Dynamics bug not Bigger Reactors.
